src/decode.rs

```diff
@@ -3525,280 +3525,231 @@ unsafe fn decode_b(
-                            15 as libc::c_int as size_t,
-                        ))
-                        .wrapping_add(1 as libc::c_int as libc::c_uint)
-                            as int8_t;
+                    let ctx = (sign_v == 2) as usize * 3 + sign_u as usize;
+                    b.cfl_alpha_mut()[1] = 1 + dav1d_msac_decode_symbol_adapt16(
+                        &mut ts.msac,
+                        (ts.cdf.m.cfl_alpha[ctx]).as_mut_ptr(),
+                        15 as libc::c_int as size_t,
+                    ) as i8;
+
                     if sign_v == 1 {
-                        b.c2rust_unnamed.c2rust_unnamed.cfl_alpha[1] =
-                            -(b.c2rust_unnamed.c2rust_unnamed.cfl_alpha[1] as libc::c_int)
-                                as int8_t;
+                        b.cfl_alpha_mut()[1] = -b.cfl_alpha()[1];
                     }
                 } else {
-                    b.c2rust_unnamed.c2rust_unnamed.cfl_alpha[1] = 0 as libc::c_int as int8_t;
+                    b.cfl_alpha_mut()[1] = 0;
                 }
+
                 if DEBUG_BLOCK_INFO(f, t) {
-                    printf(
-                        b"Post-uvalphas[%d/%d]: r=%d\n\0" as *const u8 as *const libc::c_char,
-                        b.c2rust_unnamed.c2rust_unnamed.cfl_alpha[0] as libc::c_int,
-                        b.c2rust_unnamed.c2rust_unnamed.cfl_alpha[1] as libc::c_int,
+                    println!(
+                        "Post-uvalphas[{}/{}]: r={}",
+                        b.cfl_alpha()[0],
+                        b.cfl_alpha()[1],
                         ts.msac.rng,
                     );
                 }
             } else if b_dim[2] + b_dim[3] >= 2
-                && b.c2rust_unnamed.c2rust_unnamed.uv_mode as libc::c_int
-                    >= VERT_PRED as libc::c_int
-                && b.c2rust_unnamed.c2rust_unnamed.uv_mode as libc::c_int
-                    <= VERT_LEFT_PRED as libc::c_int
+                && b.uv_mode() >= VERT_PRED as u8
+                && b.uv_mode() <= VERT_LEFT_PRED as u8
             {
-                let acdf_0: *mut uint16_t =
-                    (ts.cdf.m.angle_delta[(b.c2rust_unnamed.c2rust_unnamed.uv_mode as libc::c_int
-                        - VERT_PRED as libc::c_int)
-                        as usize])
-                        .as_mut_ptr();
-                let angle_0 = dav1d_msac_decode_symbol_adapt8(
-                    &mut ts.msac,
-                    acdf_0,
-                    6 as libc::c_int as size_t,
-                ) as libc::c_int;
-                b.c2rust_unnamed.c2rust_unnamed.uv_angle = (angle_0 - 3) as int8_t;
+                let acdf = &mut ts.cdf.m.angle_delta[b.uv_mode() as usize - VERT_PRED as usize];
+                let angle = dav1d_msac_decode_symbol_adapt8(&mut ts.msac, acdf.as_mut_ptr(), 6)
+                    as libc::c_int;
+                *b.uv_angle_mut() = (angle - 3) as int8_t;
```

Comment:
Above, this was `angle as int8_t - 3`.  And no cast on the line above.

Reply:
I'm not sure what you're referring to? I think the original version of this line was:

```rust
b.c2rust_unnamed.c2rust_unnamed.uv_angle = (angle_0 - 3) as int8_t;
```

Reply:
See https://github.com/memorysafety/rav1d/pull/203#discussion_r1183195055 actually.  I meant to comment on that one rather than this one (I had them flipped).